java/fury-core/src/main/java/org/apache/fury/resolver/ClassResolver.java

```diff
@@ -1674,6 +1674,33 @@ public ClassInfo readClassInfo(MemoryBuffer buffer) {
     return classInfo;
   }
 
+  public ClassInfo readClassInfo(MemoryBuffer buffer, Class<?> targetClass) {
+    if (metaContextShareEnabled) {
+      ClassInfo classInfo = readClassInfoWithMetaShare(buffer, fury.getSerializationContext().getMetaContext());
+
+      // replace target class if needed
+      if (!targetClass.getName().equals(classInfo.getCls().getName())) {
```

Comment:
```suggestion
      if (targetClass != classInfo.getCls()) {
```

this will be much faster